aten/src/ATen/native/native_functions.yaml

```diff
@@ -56,7 +56,8 @@
   dispatch:
     CUDA: _cudnn_rnn_backward
 
-- func: _cudnn_init_dropout_state(float dropout, bool train, int dropout_seed, TensorOptions options) -> Tensor
+- func: _cudnn_init_dropout_state(float dropout, bool train, int dropout_seed, *, ScalarType dtype=float, Layout layout=strided, Device device=\"cpu\") -> Tensor
```

Comment:
These defaults are wrong:
`dtype`: the default is the default tensor type
`layout`: strided is correct for some of these, but wrong for others (e.g. sparse_coo_tensor below has default strided which is clearly wrong)
`device`: the default device is the device of the default tensor type.

Is there a reason we can't make the defaults for all of these None if they are determined at runtime?  (It's fine to have a real default for layout but it should be correct).

Reply:
These defaults are determined based on what the JIT generates. They are indeed not logical, but that doesn't mean they're wrong in the way it's currently implemented.

Reply:
Marking this for further edits: Shouldn't have defaults